src/data.js

```diff
@@ -141,39 +141,45 @@ Namespace.ensureNamespace = function(namespace, expires) {
 };
 
 /**Delete expired entries */
-Namespace.expireEntries = function(parent, indexedTask, continuationToken=null) {
-  console.log(`expireEntries in '${parent}' with token '${continuationToken}'`);
-  return this.query({
-    parent: parent,
+Namespace.expireEntries = function(indexedTask, continuationToken=null) {
+  //console.log(`expireEntries in '${parent}' with token '${continuationToken}'`);
+  return this.scan({
   },
   {
     limit:         500,
     continuation:   continuationToken,
-  }).then(async (data) => { 
+  }).then(async (data) => {
+    console.log('..namespaces', data);
     var dataLength = data.entries.length;
+    now = new Date();
+    now = new Date(
+      now.getFullYear(),
+      now.getMonth(),
+      now.getDate(),
+      0, 0, 0, 0
+    );
+    now = now.getTime();
     
     for (var i=0; i<dataLength; i++) {
       let entry = data.entries[i];
       console.log(`..entry ${JSON.stringify(entry)}`);
-      let namespace = parent + '.' + entry.name;
-      if (parent.length === 0 || entry.name.length === 0) {
-        namespace = parent + entry.name;
+      let namespace = entry.parent + '.' + entry.name;
+      if (entry.parent.length === 0 || entry.name.length === 0) {
+        namespace = entry.parent + entry.name;
       }
-      await this.expireEntries(namespace, indexedTask);
-
       // insertTask is careful to update expires of entries 
       // from the root out to the leaf. A parent's expires is
       // always later than the child's. Hence, we can delete an
       // entry without checking its children.
       await indexedTask.expireTasks(namespace);
```

Comment:
I suggest calling it from within `Namespace.expireEntries` , although outside of the for loop to ensure a full table scan once. This is because a removal of a namespace should imply removal of all tasks indexed under that namespace. 
I can also call it directly from the test method if you like :)